_studio/mfx_lib/shared/include/mfx_h264_enc_common_hw.h

```diff
@@ -837,53 +837,73 @@ namespace MfxHwH264Encode
             mfxExtBuffer * p = MfxHwH264Encode::GetExtBuffer(
                 m_extParam,
                 m_numExtParam,
-                ExtBufTypeToId<typename GetPointedType<T>::Type>::id);
+                ExtBufTypeToId<typename GetPointedType<T>::Type>::id,
+                m_fieldId);
             return reinterpret_cast<T>(p);
         }
 
         template <typename T> friend mfxExtBufferProxy GetExtBuffer(const T & par);
+        template <typename T> friend mfxExtBufferProxy GetExtBuffer(const T & par, mfxU32 fieldId);
```

Comment:
del

Reply:
set default value (fieldId = 0) in declaration leads Android build error, in definition - desktop build error

Reply:
@akokoshn what is the build error?

Reply:
if set fieldId = 0 in **declaration**
**Android build** - "friend declaration specifying a default argument must be a definition template <typename T> friend mfxExtBufferProxy GetExtBuffer(const T & par, mfxU32 fieldId = 0);"
in **definition**
**desktop build** - "redeclaration of ‘template<class T> MfxHwH264Encode::mfxExtBufferProxy MfxHwH264Encode::GetExtBuffer(const T&, mfxU32)’ may not have default arguments [-fpermissive]"

Reply:
@akokoshn hm, ok if compiler saying about redeclarations it should mention where previous declaration was found. Maybe we already have some aliases in code

Reply:
@mgonchar , seems re declaration error cause by template limitation: https://stackoverflow.com/questions/32105975/why-cant-i-have-template-and-default-arguments
It's may be avoid by adding "-fpermissive" to CMAKE_CXX_FALAGS
But it's not avoid android build issue: "friend declaration specific a default argument must be a definition"

What wrong with overloads functions? I guess in case of using friend declaration and template overloads are more understandable

Reply:
Please, do NOT introduce -fpermissive!